### ISSUE TITLE
file_url should be replaced by direct LFN in core.File class initializer

### ISSUE THREAD

**Issue by ahnitz** [ligo-cbc/pycbc#297]
I think this is a relic of when core.File was a subclass of cache entry and not pegasus.File.

We provide two methods for creating a file object, one that allows for lazy naming (i.e implicit), and one where we provide the name as it is either chosen for us by already existing, or cannot be arbitrary. 

The confusion is that although we accept a file_url, it really does have to be a file:// url and not something else. We derive an LFN from it, and also the storage path. I think we should instead have the user provide only an LFN, and in the case that the directory option in conjuction with store_file=Ture, we use that for the storage path. This separates the LFN and storage path. 

PFNs, as are now, are always given after creating the File object by using the standard Pegasus.DAX3.File method.

The key thing is that I think this would make it a lot easier to allow arbitrary file urls (like gsiftp, http) in the configuration file, and cleans up all the places we do things like this. 

https://github.com/ligo-cbc/pycbc/blob/master/pycbc/workflow/tmpltbank.py#L325

**Comment by spxiwh** [comment]
I agree Alex. As this is not critical, and has the potential to cause workflow generation scripts to fail if we miss some places, I would suggest to hold off addressing this for a week or two.

Cheers
Ian

**Comment by duncan-brown** [comment]
Larne, please can you look into this and prepare a patch to clean up the code?

**Comment by duncan-brown** [comment]
Removing from O1 critial per discussion with @spxiwh but @lppekows is still working on this.
